### PR TITLE
`azurerm_shared_image_version` - Support `replicated_region_deletion_enabled` and `target_region.exclude_from_latest_enabled`

### PR DESCRIPTION
**Pull request by ms-zhenhua**
## Swagger Link
https://github.com/Azure/azure-rest-api-specs/blob/d99cad13db6f16e55ddf4e45879980c9e1aaf178/specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2022-03-03/gallery.json#L2838

https://github.com/Azure/azure-rest-api-specs/blob/d99cad13db6f16e55ddf4e45879980c9e1aaf178/specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2022-03-03/gallery.json#L2454C10-L2454C27

## Test Result

image

### CODE REVIEW

File: internal/services/compute/shared_image_version_resource.go [hunk 1]
```diff
@@ -177,6 +184,13 @@ func resourceSharedImageVersion() *pluginsdk.Resource {
 				Default:  false,
 			},
 
+			"replicated_region_deletion_enabled": {
```
Comment (tombuildsstuff) [inline internal/services/compute/shared_image_version_resource.go:187]: tbh this'd be clearer/closer to the original name whilst still consistent as:

```suggestion
			"deletion_of_replicated_locations_enabled": {
```

File: website/docs/r/shared_image_version.html.markdown [hunk 2]
```diff
@@ -95,6 +97,8 @@ The `target_region` block supports the following:
 
 * `disk_encryption_set_id` - (Optional) The ID of the Disk Encryption Set to encrypt the Image Version in the target region. Changing this forces a new resource to be created.
 
+* `exclude_from_latest_enabled` - (Optional) Whether excluding this Image Version from the `latest` filter. If set to `true` this Image Version won't be returned for the `latest` version. Defaults to `false`.
```
Comment (tombuildsstuff) [inline website/docs/r/shared_image_version.html.markdown:100]:
```suggestion
* `exclude_from_latest_enabled` - (Optional) Specifies whether this Shared Image Version should be excluded when querying for the `latest` version. Defaults to `false`.
```

File: website/docs/r/shared_image_version.html.markdown [hunk 3]
```diff
@@ -77,6 +77,8 @@ The following arguments are supported:
 
 -> **NOTE:** You must specify exact one of `blob_uri`, `managed_image_id` and `os_disk_snapshot_id`.
 
+* `replicated_region_deletion_enabled` - (Optional)  Whether removing this Gallery Image Version from replicated regions is allowed. Defaults to `false`. Changing this forces a new resource to be created.
```
Comment (tombuildsstuff) [inline website/docs/r/shared_image_version.html.markdown:80]: Presumably this is?

```suggestion
* `deletion_of_replicated_locations_enabled` - (Optional) Specifies whether this Shared Image Version can be deleted from the Azure Regions this is replicated to. Defaults to `false`. Changing this forces a new resource to be created.
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @ms-zhenhua 

Thanks for this PR - I've taken a look through and left some comments inline, but if we can fix those up then we should be able to take another look and get this merged.

Thanks!

File: internal/services/compute/shared_image_version_resource.go (hunk 1 above)
Comment (ms-zhenhua) [reply internal/services/compute/shared_image_version_resource.go]: code updated. Thanks.

File: website/docs/r/shared_image_version.html.markdown (hunk 2 above)
Comment (ms-zhenhua) [reply website/docs/r/shared_image_version.html.markdown]: doc updated. Thanks.

File: website/docs/r/shared_image_version.html.markdown (hunk 3 above)
Comment (ms-zhenhua) [reply website/docs/r/shared_image_version.html.markdown]: doc updated. Thanks.

**Comment by ms-zhenhua** [comment]
Hi @tombuildsstuff, thank you for your review. I have updated the code and doc. Could you kindly take another review? Thanks.

## Test Result
=== RUN   TestAccSharedImageVersion_replicatedRegionDeletion
=== PAUSE TestAccSharedImageVersion_replicatedRegionDeletion
=== CONT  TestAccSharedImageVersion_replicatedRegionDeletion
--- PASS: TestAccSharedImageVersion_replicatedRegionDeletion (1205.70s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/compute       1235.581s

**Review by stephybun — APPROVED** [review]
Thanks @ms-zhenhua. LGTM 🥞

**Pull request merged**